imap/caldav_alarm.c

```diff
@@ -2179,20 +2179,20 @@ EXPORTED int caldav_alarm_process(time_t runtime, time_t *intervalp, int dryrun)
                 num_user_records = 0;
                 skip_user = 0;
                 free(userid);
-                mboxname_release(&nslock);
+                user_nslock_release(&user_nslock);
                 userid = xstrdup(mbname_userid(mbname));
                 if (user_isreplicaonly(userid)) {
                     skip_user = 1;
                     continue;
                 }
-                nslock = user_namespacelock_full(userid, LOCK_NONBLOCKING);
+                user_nslock = user_nslock_full1(userid, LOCK_NONBLOCKING);
             }
             mbname_free(&mbname);
 
             if (skip_user) continue;
 
             // if we failed to lock the user, or have done too many for this user, skip
-            if (!nslock || ++num_user_records > MAX_CONSECUTIVE_ALARMS_PER_USER) {
+            if (user_nslock || ++num_user_records > MAX_CONSECUTIVE_ALARMS_PER_USER) {
```

Comment:
Did we loose the negation here, should this be `!user_nslock`?

Reply:
Funny enough, guess what!  Calalarmd failed to process anything because... this negation was lost!